src/mds/QuiesceDbManager.cc

```diff
@@ -82,7 +82,7 @@ void* QuiesceDbManager::quiesce_db_thread_main()
         std::chrono::seconds(10)
     );
 
-    while (!db_thread_has_work()) {
+    while (!db_thread_has_work() || bootstrap_delay.has_value()) {
```

Comment:
It's not going to work. There are cases when the bootstrap delay will be set but we should not wait because there are important things to process, like the membership update or peer responses in the form of the db listing.

Please consider changing the `db_thread_has_work` so that you pass the `bootstrap_delay.has_value()` to it as an argument, and then there for every possible thing that could wake up our thread you should also consider if we're currently in the bootstrap delay mode. Specifically, you'll want to ignore non-empty acks and requests in that function, but nothing else.

Reply:
> and then there for every possible thing that could wake up
can you elaborate on this please.

Reply:
> you should also consider if we're currently in the bootstrap delay mode

the `bootstrap_delay.has_value()` makes sure of it right?

Reply:
here's my [message](https://ceph-storage.slack.com/archives/C04LVQMHM9B/p1719317503435309?thread_ts=1719230835.852789&cid=C04LVQMHM9B) from slack on the topic

> and alternative approach - a more robust one - would be to never sleep anywhere else but at the beginning of the main loop, but also consider the bootstrapping condition there, so that we ignore pending acks and requests. That could be an argument to `db_thread_has_work(bool leader_bootstrapping = false)`

Just look at that function. I think you have enough info. Please continue working on this PR, I'll be available for a review